### PR TITLE
HOST networking breaks ports for applications created via PUT

### PR DESCRIPTION
**Pull request by meln1k**
Summary: change in the appConversion class for correct transformation of empty seq to option

JIRA issues: MARATHON-7969

### CODE REVIEW

**Comment by jdef** [comment]
I'd actually like to see a test case that exercises PUT in the manner suggested by the JIRA report, so that I understand the resulting application that Marathon ends up creating. The original JIRA talks about `PORT` and `portMappings` - it's not immediately clear to me how either of those are related to this PR.

**Comment by jdef** [comment]
The sample JSON from the JIRA (when PUT'ing as a new app):
```json
{
  "id":"tomcat",
  "mem":512,
  "cpus":1.0,
  "instances":1,
  "container": {
    "type":"DOCKER",
    "docker": {
      "image":"tomcat:8.0",
      "network":"HOST"
    }
  }
}
```
... should yield an app w/ a single port definition: `{ "port": 0 }` -- that's the test that I'd like to see

**Comment by meln1k** [comment]
@jdef the problem that we can only test it in integration test, because unit tests with PUT uses mocked implementation. I tested it manually and it works.

@timcharper do you think we should have an integration test here?

File: src/test/scala/mesosphere/marathon/raml/AppConversionTest.scala [hunk 1]
```diff
@@ -94,6 +94,15 @@ class AppConversionTest extends UnitTest with ValidationTestLike {
     behave like convertToRamlAndBack(hostApp)
     behave like convertToProtobufThenToRAML(hostApp)
 
+    "correctly handle empty port definitions" in {
+      Given("An app definition")
```
Comment (timcharper) [inline src/test/scala/mesosphere/marathon/raml/AppConversionTest.scala]: With such a small test, I'm inclined to believe that these given, then, when statements add very little value. Let's remove them?

Comment (timcharper) [reply src/test/scala/mesosphere/marathon/raml/AppConversionTest.scala]: Also, in the links JIRA, it seems that port definitions wasn't specified at all. I think that the JIRA could be potentially clarified.

What we should really be testing is AppUpdate, right?

**Comment by alenkacz** [comment]
@meln1k just out of curiosity why did we close this? I did not find info not here not in JIRA... :)

**Comment by meln1k** [comment]
@alenkacz it turned out that fixing this bug will consume much more time that was initially planned.

**Comment by jdef** [comment]
> it turned out that fixing this bug will consume much more time that was initially planned.

Can you please write up an RCA somewhere to help the next person that tackles this?

**Comment by alenkacz** [comment]
@jdef @meln1k yeah that would be nice - I think JIRA is the right place for that...

**Pull request closed without merging**